packages/flutter_tools/lib/src/run_cold.dart

```diff
@@ -157,8 +167,12 @@ class ColdRunner extends ResidentRunner {
   @override
   void printHelp({ @required bool details }) {
     bool haveDetails = false;
-    if (_result.hasObservatory)
-      printStatus('The Observatory debugger and profiler is available at: ${_result.observatoryUri}');
+    for (RunningDevice device in runningDevices) {
+      if (device.observatoryUris != null) {
+        for (Uri uri in device.observatoryUris)
+          printStatus('An Observatory debugger and profiler is available at $uri');
```

Comment:
This is going to look really ugly, just saying the same thing over and over with different ports, and with no way for the user to tell which one they should use.

Can we improve this at all in the multi-device case?